src/storage/txn/actions/shared.rs

```diff
@@ -72,9 +79,37 @@ pub(super) fn prewrite_key_value<S: Snapshot>(
 
         txn.guards.push(key_guard);
     }
-    txn.put_lock(key, &lock);
+
+    if try_one_pc {
+        txn.put_locks_for_1pc(key, lock, has_pessimistic_lock);
+    } else {
+        txn.put_lock(key, &lock);
+    }
 
     fail_point!("after_prewrite_one_key");
 
-    Ok(async_commit_ts)
+    Ok(final_min_commit_ts)
+}
+
+pub(in crate::storage::txn) fn handle_1pc<S: Snapshot>(
+    txn: &mut MvccTxn<S>,
+    commit_ts: TimeStamp,
+) -> (TimeStamp, ReleasedLocks) {
```

Comment:
The returned timestamp seems unncessary because it always equals to  `commit_ts`.

Reply:
Oh, yes.